src/main/java/seedu/duke/commands/UpdateCommand.java

```diff
@@ -26,60 +35,76 @@ public UpdateCommand(String arguments) throws DukeException {
         this.flags = Arrays.copyOfRange(details, 1, details.length);
     }
 
+    /**
+     * A core logic method for updating the current food attributes with given flag values
+     *
+     * @param currentFood
+     * @param flagName
+     * @param flagValue
+     * @throws DukeException
+     */
+    private void updateFoodAttribute(Food currentFood, String flagName, String flagValue) throws DukeException {
+        try {
+            switch (flagName) {
+            case "n":
+                currentFood.setName(flagValue);
+                break;
+            case "e":
+                Validator.isExpiryDateValid(flagValue);
+                currentFood.setExpiryDate(flagValue);
+                break;
+            case "q":
+                if (currentFood.getQuantity() == 0 && Arrays.asList(flags).contains("u")) {
+                    throw new DukeException("Can't set quantity with no unit provided");
+                }
+                currentFood.setQuantity(Double.parseDouble(flagValue));
+                String currentFoodUnit = currentFood.getUnit();
+                currentFood.setUnit(currentFoodUnit);
+                break;
+            case "u":
+                currentFood.setUnit(flagValue);
```

Comment:
do u want to throw a duke exception here something like this. so that --u can only be set when there is a quantity if not something weird will show (i show example in the next comment picture)
![image](https://user-images.githubusercontent.com/88308976/230635767-bc8dd8ce-43f5-4a1b-a2eb-130dd6253bb6.png)

Reply:
![image](https://user-images.githubusercontent.com/88308976/230636163-09aa04c7-6de9-457a-a290-e8831013b4c9.png)

Reply:
Hey @wanjuin, why do you find it weird? The unit can be set without without quantity and I think it makes sense. I also think it makes sense that it shows nothing until the quantity is more than zero. If you add a unit then quantity you will see that everything is fine, too.
Here is the screenshot 
![image](https://user-images.githubusercontent.com/52950994/230701136-67c420cc-da10-4e3d-81eb-edef71f6dacc.png)

Reply:
Perhaps users can update multiple attributes in one command line instead of only one change in one command line.

Reply:
@tsx0314, you can update several things in the update command ofc, you can try it)

Reply:
I changed the `toString` method a little bit, we can improve this thing in another PR if some other great idea appear, cause it's not just about update

Reply:
I think the updating part is okay. As for the Food Class `toString` part, I set it as if `quantity == 0`, it will not show quantity and unit; else it will print quantity and units. However, users can only check the feedback message and they may feel that the unit is not updated as there is no sign showing that it is updated.

Perhaps you can add another message to tell the user that “As you are not adding any quantity, we will not show the quantity and updated unit”  or a feedback message like “You have successfully updated `unit` to `kg`!” to avoid misunderstanding!